src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs

```diff
@@ -208,22 +212,50 @@ private IFileProvider GetFileProvider(IServiceProvider requestServices)
             return FileProvider;
         }
 
-        private static async Task CopyStreamToResponse(
-            string fileName,
+        private Task CopyPhysicalFileToResponseAsync(
+            HttpResponse response,
+            string physicalFilePath,
+            CancellationToken cancellationToken)
+        {
+            var sendFile = response.HttpContext.GetFeature<IHttpSendFileFeature>();
+            if (sendFile != null)
+            {
+                return sendFile.SendFileAsync(
+                    physicalFilePath,
+                    offset: 0,
+                    length: null,
+                    cancellation: cancellationToken);
+            }
+            else
+            {
+                var fileStream = new FileStream(
```

Comment:
@davidfowl : I have verified this now and it works fine with read-only files too.